libgnucash/engine/kvp-frame.hpp

```diff
@@ -247,7 +247,7 @@ void KvpFrame::for_each_slot_prefix(std::string const & prefix,
                 return;
             /* Testing for prefix matching */
             if (std::mismatch(prefix.begin(), prefix.end(), temp_key.begin()).first == prefix.end())
-                func (a.first, a.second);
+                func (&a.first[prefix.size()], a.second);
```

Comment:
This kind of expression in C++ makes me suspicious always (the `first[prefix.size()]` part). How can you guarantee that the container `a.first` has enough elements so that `prefix.size()` is not out of bounds? If there is a possibility to write this differently (such as using `back()`), please change that. Or add a comment for why this is ok here.

Reply:
@cstim It's guaranteed that by line 245, which returns if `strlen(a.first) < prefix.size()`, but what you're complaining about isn't part of this PR, it's existing code.

@gruberchr If you want a performance win don't create a std::string just to use operator==(), use `if (strncmp(a.first, prefix.c_str(), prefix.size()))`. Even if the std::string ctor doesn't allocate it's still unnecessary overhead and I doubt that std::mismatch is any less efficient than either strncmp or std::string::operator==(); they all have to iterate over the char arrays and compare each until there's a mismatch. Any performance hit is still much more likely to be from the std::string ctor and if you saw an improvement it would be because with your test data the substring was able to avoid an alloc because it fit in your std library's SSTO window and the whole a.first didn't.

Reply:
@jralls I didn't measure performance. I simply tried to exchange `std::mismatch` with string comparison as you suggested in the corresponding bug ticket. And AFAIKS string comparison uses `std::memcmp` internally, which seems to be faster than `std::mismatch`. Since `strncmp` has comparable performance than `memcmp`, the only performance difference between using C++ and C for string comparison should be the ctor overhead as you already mentioned.
I'll change this using C string comparison instead of C++ string comparison.

Reply:
Any further remarks?

Reply:
Yeah, this template should be removed, it's not used anywhere--even in tests. Only the three-argument version is used.

Reply:
Done